### PR TITLE
[21] Contact Form Validation

### PR DESCRIPTION
**Pull request by faithakinbobola**
## Task: `[21] Contact Form Validation`
- `I used DOM on the Contact Form, passed an error message if the inputs are empty or doesn't match the regular expression.`

### CODE REVIEW

File: submissions/akinbobolaFaith/dom/contactForm.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    fullNameValidation(e);
+    fieldValidation(e, email, emailRegex, "Please enter your Email Address", "Please enter a valid mail");
+    fieldValidation(e, subject, fieldRegex, "Please enter a message subject");
+    fieldValidation(e, message, fieldRegex, "Please enter your Message");
+    return true;
+}
+
+sendForm.addEventListener("submit", informate);
+;
```
Comment (ToluAkin) [inline submissions/akinbobolaFaith/dom/contactForm.js:50]: @faithakinbobola Remove the semicolon here.

**Review by ToluAkin — CHANGES_REQUESTED** [review]

**Review by ToluAkin — APPROVED** [review]

File: submissions/akinbobolaFaith/dom/contactForm.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+}
+
+function fieldValidation(event, field, fieldRegex, innerHTML, innerHTMLTwo) {
+    if (field.value === '') {
+        field.nextElementSibling.innerHTML = innerHTML;
+        event.preventDefault();
+        return false;
+    } else if (!field.value.match(fieldRegex)) {
+        field.nextElementSibling.innerHTML = innerHTMLTwo || "";
```
Comment (adaugochi) [inline submissions/akinbobolaFaith/dom/contactForm.js:30]: What does `innerHTMLTwo` do?

File: submissions/akinbobolaFaith/dom/contactForm.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+}
+
+function informate(e) {
+    const emailRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,6})+$/;
+    const fieldRegex = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[._$,/;:+=~`'"<>^#?!@$%^&*-])$/;
+
+    fullNameValidation(e);
+    fieldValidation(e, email, emailRegex, "Please enter your Email Address", "Please enter a valid mail");
+    fieldValidation(e, subject, fieldRegex, "Please enter a message subject");
```
Comment (adaugochi) [inline submissions/akinbobolaFaith/dom/contactForm.js:44]: Also why the regex for `message` and `subject` field. I don't think that is part of the task requirements

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/akinbobolaFaith/dom/contactForm.js (hunk 2 above)
Comment (faithakinbobola) [reply submissions/akinbobolaFaith/dom/contactForm.js:30]: It represents the second criteria for the email address field

File: submissions/akinbobolaFaith/dom/contactForm.js (hunk 3 above)
Comment (faithakinbobola) [reply submissions/akinbobolaFaith/dom/contactForm.js:44]: I should remove it then

**Pull request merged**